### DISCUSSION TITLE
apiman with mariadb databse

### DISCUSSION THREAD

**Discussion by Satishshelake** [apiman/apiman#2659] [Help]
Hello,
Does anyone used mariadb database for apiman? I am new in it. DO we have any blogs avaible for same to changes?

**Comment by msavy** [comment]
I suggest you use Postgres for now. If you are a support customer feel free to get in touch, but otherwise you'll probably need to wait for Apiman 4 and I can't promise anything. I have clients successfully running on Microsoft SQL Server, etc, so it's definitely possible with some engineering effort.

**Comment by Satishshelake** [comment]
Recursive sql 'WITH' statement is not supported in triggers mariadb, chanaged triggers.mysql.sql file. Triggers are working now, But couldn't managed error  Caused by: org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: wrong column type encountered in column [payload] in table [notifications]; found [longtext (Types#LONGVARCHAR)], but expecting [json (Types#OTHER)]
17:30:37,770 ERROR [stderr] (Thread-112)        at org.hibernate@5.3.20.Final//org.hibernate.tool.schema.internal.AbstractSchemaValidator.validateColumnType(AbstractSchemaValidator.java:159)


Also tried alter in liquibase configuration.
